File: src/CSPAF/CSPAgents/CSPAgentDetails/CSPUpdateAgent.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace CSPAF.CSPAgents.CSPAgentDetails
{
    class CSPUpdateAgent
    {
        public void run(Dictionary<string, object> Params)
        {
            var version = (string)Params["Version"];
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPAgentDetailsController)navigationController.VisibleViewController;
            controller.UpdateAgent(version);
        }
    }
}

File: src/CSPAF/CSPAgents/CSPAgentDetails/CSPUpdateAgent.java
package CSPAF.CSPAgents.CSPAgentDetails;

import java.io.*;
import java.util.*;

import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

class CSPUpdateAgent
{
    public void run(HashMap<String, Object> params) throws IOException
    {
        String version = params.get("Version").toString();
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPAgentDetailsController controller = (CSPAgentDetailsController)navigationController.getVisibleViewController();
        controller.updateAgent(version);
    }
}